fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/domain/CustomJobParameterRepository.java

```diff
@@ -18,6 +18,11 @@
  */
 package org.apache.fineract.infrastructure.jobs.domain;
 
-import org.springframework.data.jpa.repository.JpaRepository;
+import java.util.Optional;
 
-public interface CustomJobParameterRepository extends JpaRepository<CustomJobParameter, Long> {}
+public interface CustomJobParameterRepository {
+
+    Long save(String jsonString);
```

Comment:
I think responsibilities are not enforced here.
The repository should take care of its own representation (JSON) and not expose that to outside callers.

If we're saying the CustomJobParameter should always be a List of JobParameterDTO, then let's force that in the interface as well.

Instead of accepting a JSON string, let's use a List<JobParameterDTO> and inside the implementation, take care of the serialization into JSON.

That way, we are making sure that nobody is saving an incorrect JSON structure in the DB for the CustomJobParameters.

Reply:
Fixed